src/priority_queue.js

```diff
@@ -0,0 +1,105 @@
+export default class PriorityQueue {
+  constructor() {
+    let _items = []
+    let _length = 0
+
+    this.getLength = function() {
+      return _length
+    }
+    this.incrementLength = function() {
+      _length++
+    }
+    this.decrementLength = function() {
+      _length--
+    }
+
+    this.getItems = function() {
+      return _items
+    }
+    this.setItems = function( newItems ) {
+      _items = newItems
+    }
+  }
+
+  sort( items ) {
+    let length = this.getLength()
+
+    for( let i = length - 1; i > -1; i-- ) {
+      for( let x = 1; x <= i; x++ ) {
+        if ( items[ x-1 ].priority < items[ x ].priority ) {
+          let temp = items[ x - 1 ]
+          items[ x - 1 ] = items[ x ]
+          items[ x ] = temp
+        }
+      }
+    }
+    this.setItems( items )
+  }
+
+  resetItemKeys( items ) {
+    let index = 0
+    let newItems = items.reduce( ( acc, item ) => {
+      acc[ index ] = item
+      index++
+      return acc
+    }, [] )
+    return newItems
+  }
+
+  enqueue( item, priority ) {
+    let items = this.getItems()
+    let length = this.getLength()
+    let node = new Node( item, priority )
+
+    items[length] = node
+    this.incrementLength()
+    this.sort( items )
```

Comment:
Yikes this is resource intensive! If your curious about optimization, you might want to look into something called "Binary Insert" which performs it's own sorting upon insertion, generally faster under certain circumstances that are beyond my understanding. 
http://machinesaredigging.com/2014/04/27/binary-insert-how-to-keep-an-array-sorted-as-you-insert-data-in-it/